fix: detect smart contracts for every evm chains

NeOMakinG opened the pull request:
## Description

The `useIsSmartContractAddress` address hook and `isSmartContractAddress` helper should take an non-optional chainId parameter because we need to take in account the multichain nature on features such as the swapper 

<!-- Please describe your changes -->

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

<!-----------------------------------------------------------------------------
If applicable, please link to the github issue and put `closes #XXXX` in your comment to auto-close the issue that your PR fixes.
------------------------------------------------------------------------------>

closes #7328

## Risk
⚠️  High, this PR already did cause us to revert because it was blocking a few trades such as RUNE => BTC

## Testing

With a smart contract wallet such as gnosis safe, and with another classic wallet such as metamask:

- Please be sure that the swapper is entirely working as expected, including non EVM trades/bridges such as RUNE => BTC
- use the swapper and verify that the from asset trigger the manual destination address input if it's a smart contract wallet or not depending if the wallet is not a smart contract
- Thorchain savers works as expected (deposit, withdraw
- Fox voting power is working as expected
- Thorchain deposit liquidity into pool is working as expected
- Loan repay (I couldn't test as I don't have any loans)

### Engineering

<!-----------------------------------------------------------------------------
Include sufficient information here for an engineer to test your PR. This may include how to test locally, in a built environment, changes to infrastructure etc.
------------------------------------------------------------------------------>

### Operations

<!-----------------------------------------------------------------------------
If your changes have a user-facing impact, describe how a non-technical QA team can functionally test your changes in a preview environment.

If they are not user-facing please describe how to test for any regressions that may occur.
------------------------------------------------------------------------------>

## Screenshots (if applicable)

## Safe wallet with avalanche
![image](https://github.com/shapeshift/web/assets/14963751/cfa4a768-4f06-49eb-aecc-b5d61fc35049)

## Safe wallet with ethereum
![image](https://github.com/shapeshift/web/assets/14963751/657fc45a-3482-4269-8f55-8d5ebeea2850)

## Another wallet
![image](https://github.com/shapeshift/web/assets/14963751/7f858c98-48a6-4e6d-bd49-4204b64cf775)

## Ethereum
![image](https://github.com/shapeshift/web/assets/14963751/8c09290f-8734-418f-adc1-f756b0c6d82b)

## Thorchain smart contract wallet not supported
![image](https://github.com/shapeshift/web/assets/14963751/845da4de-d92e-404b-8193-b675b5ab19eb)

## Add Liquidity smart contract wallet not supported
![image](https://github.com/shapeshift/web/assets/14963751/a97d31a7-161a-4384-9aa5-bc0c657078f0)

## add liquidity non-smart contract address
![image](https://github.com/shapeshift/web/assets/14963751/b129fad9-2e93-4e8a-93af-3108c46b55ed)

## Savers deposit confirm smart contract address
![image](https://github.com/shapeshift/web/assets/14963751/c5089087-769d-4df4-bf03-2becc0a306e8)

## Savers deposit confirm classic wallet
![image](https://github.com/shapeshift/web/assets/14963751/2c680cb2-2d74-49fb-8339-4d419d4bebc9)

## Withdraw savers confirm smart contract address
![image](https://github.com/shapeshift/web/assets/14963751/8ba1ee9c-d859-414c-b3e5-e1116b1dc0ba)

## Withdraw savers deposit confirm classic wallet
I couldn't test because I don't have any position on this wallet and not enough funds

## Fox voting power smart contract wallet
I couldnt test as I have no FOX on my smart contract wallet

## Fox voting power with classic wallet
![image](https://github.com/shapeshift/web/assets/14963751/fa383655-5bfa-4183-9214-45008e83dbbb)

## Rune => Avax
![image](https://github.com/shapeshift/web/assets/14963751/93751bd4-1112-4fd2-8bd8-7b089303bf43)

## Rune => BTC
![image](https://github.com/shapeshift/web/assets/14963751/804c8644-8692-4161-9ddc-9b874be9a24d)

woodenfurniture reviewed (approved):
Everything appears to be working, including fox discount simulation on gnosis safe wallet. Lending was untested due to pools being full and no current positions.

kaladinlight reviewed (changes requested):
![image](https://github.com/user-attachments/assets/07e65e0d-9d7a-4a8e-afb8-0ea891e6e6b2)

Add Liquidity Input is checking for smart contract interaction and showing error state, but is NOT disabling the button.

kaladinlight commented:
A couple follow up questions:
- Are both `isLoading` and `isFetching` necessary? Haven't seen much use of `isFetching` and just curious the rationale
- Do we actually want to use the loading/fetching state as part of the disabled logic? I would think we would use those for the loading state of the button and then disable only on the isSmartContract value being `true` basically. Doesn't really change behavior (if also being used for loading state), but could be less lines of code in the disabled memo at least haha

NeOMakinG commented on src/components/MultiHopTrade/components/TradeInput/components/ConfirmSummary.tsx:
```diff
@@ -107,22 +107,22 @@ export const ConfirmSummary = ({
 
   const disableSmartContractSwap = useMemo(() => {
     // Swappers other than THORChain shouldn't be affected by this limitation
     if (activeSwapperName !== SwapperName.Thorchain) return false
```
Is this something we really want now that we give the user the control of the receive address?

kaladinlight replied:
https://github.com/shapeshift/web/pull/7277 there was this PR to remove this restraint. pretty out of date now, but regardless, we should follow up with that change separately

NeOMakinG commented:
> A couple follow up questions:
> 
> * Are both `isLoading` and `isFetching` necessary? Haven't seen much use of `isFetching` and just curious the rationale
> * Do we actually want to use the loading/fetching state as part of the disabled logic? I would think we would use those for the loading state of the button and then disable only on the isSmartContract value being `true` basically. Doesn't really change behavior (if also being used for loading state), but could be less lines of code in the disabled memo at least haha

big 🧠 kev, this is a sign that I should stop working late in the night after our retro/meeting

- isLoading is enough in that case because the only query key that could cause a refetch is the chainId, meaning that `isLoading` is enough and will be refired itself if the chainId change and hasn't been fetch in the past, I removed all those `isFetching` calls in [4dc06f1](https://github.com/shapeshift/web/pull/7358/commits/4dc06f1442425c75b0b318794b82fbbcc716cd17)
- This is true, I removed all those checks in `isDisabled` occurences in [54afc21](https://github.com/shapeshift/web/pull/7358/commits/54afc21f202158b744d1fe0de85874058b35740e)
- Also initially in [4dc06f1](https://github.com/shapeshift/web/pull/7358/commits/4dc06f1442425c75b0b318794b82fbbcc716cd17) I removed all the loading checks in useMemo where we verify that this is a smart contract or not because it's handled in the end button every time

Thanks for the super valuable review @kaladinlight , can I ask you to do a deep test again? I performed the same tests than submitted in the initial review and everything looks good but as we changed pieces of code at every place I don't feel so confident now!

kaladinlight commented on src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx:
```diff
@@ -671,6 +670,7 @@ export const BorrowInput = ({
                   quoteErrorTranslation ||
                   isEstimatedFeesDataError ||
                   isEstimatedFeesDataLoading ||
+                  isAddressByteCodeLoading ||
```
move to `isLoading`

NeOMakinG replied:
done in [02b8f00](https://github.com/shapeshift/web/pull/7358/commits/02b8f00e75f94ff8231733ed9f5ed61b8627c0d6)

kaladinlight reviewed (approved):
Noting that for both Savers and LP, our current flow will allow a smart contract address to approve first before warning them that they can't interact at confirm step. Non blocking as this behavior hasn't changed from what it was already doing and the restriction will be removed soon anyways.
